core/app/models/spree/stock/content_item.rb

```diff
@@ -12,7 +12,7 @@ def initialize(inventory_unit, state = :on_hand)
         delegate :line_item,
                  :quantity,
                  :variant, to: :inventory_unit
-        delegate :price, to: :variant
+        delegate :price, to: :line_item
```

Comment:
originally price was fetched from the variant, which pointed always to its default price, so calculations was incorrect or just causing 500 if that price(in store's default currrency) wasn't present

Reply:
flat percent shipping calculations: https://github.com/spree/spree/blob/960c9303b2d5e7838bcbfd0b41d00219d119967e/core/app/models/spree/shipping_calculator.rb#L18

default price: https://github.com/spree/spree/blob/960c9303b2d5e7838bcbfd0b41d00219d119967e/core/app/models/concerns/spree/default_price.rb#L11